src/tools/pub/pubspec.md

```diff
@@ -410,6 +414,38 @@ Each download of the package includes all screenshot files.
 Pub.dev generates the package's thumbnail image from the first screenshot. If 
 this screenshot uses animation, pub.dev uses its first frame.
  
+### Topics
+
+Package authors can use the `topics` field to categorize their package. Topics
+can be used to assist discoverability during search with filters on pub.dev.
+Pub.dev displays the topics on the package page as well as in the search
+results.
+
+The field consists of a list of names. For example:
+
+```yaml
+topics:
+  - network
+  - http
+```
+
+Pub.dev limits topics to the following specifications:
+
+- At most 5 topics per package.
+- The length of the topic name should be 2-32 characters.
+- The topic name should consist of lowercase alphanumeric characters or dash
+  (but no double dash), starting with a-z and ending with a-z or 0-9.
+
+
+#### Best Practices
+
+When choosing topics, first look at [existing topics]({{site.pub}}/topics) to
+check if established topics exist that match your package.
+
+If you have a number of related packages, tag of all these packages with the
+same topic. Such packages could belong to the same framework and be tagged with
+the name of the framework.
+
```

Comment:
```suggestion
When choosing topics it is good practice to look at
[existing topics]({{site.pub}}/topics) to see if there are established topics
that match your package.

Another good idea, if you have a number of related packages, for instance
belonging to the same framework, it is good practice to tag of all these
packages with the same topic, e.g. the name of the framework.
```
I suggest we go back to the original proposal.

--------------

```
#### Best Practices
```
We don't do this other places on this page.


> When choosing topics, first look at [existing topics]({{site.pub}}/topics) to
> check if established topics exist that match your package.

This is an instruction and not a recommendation or suggestion.
The original intent here is to suggest that users consider existing topics.


> If you have a number of related packages, tag of all these packages with the
> same topic. Such packages could belong to the same framework and be tagged with
> the name of the framework.

Again, we should suggest not instruct. Also the part about frameworks does not make sense to me.

Reply:
If we wanted a different text I'd suggest:

```suggestion
When publishing a package consider if [existing topics]({{site.pub}}/topics)
are relevant. Tagging with existing topics helps users discover your package.

If you have a number of related packages, for example packages making up a framework,
consider establishing a new topic linking these packages together. This makes it easy
to discover related packages, and allows other package authors to tag their packages
as being relevant for your framework. For example, check out
[`topic:shelf`](https://pub.dev/packages?q=topic%3Ashelf).
```

We can also omit these lines entirely and just put the recommendations in `publishing.md`.

Reply:
\cc @atsansone and @szakarias for thoughts on content and text

I think you are right though that these extensive details don't belong here and could be a section in [Publishing](https://dart.dev/tools/pub/publishing).